src/components/App.tsx

```diff
@@ -393,10 +393,15 @@ export class App extends React.Component<AppProps, AppState> {
     }
     this.setState({ fiddle });
   }
-  async gist() {
+  async gist(fileOrDirectory?: File) {
     this.logLn("Exporting Project ...");
-    const projectModel = this.state.project.getModel();
-    const gistURI = await Service.exportProjectToGist(projectModel, this.state.fiddle);
+    let gistURI: string;
+    if (fileOrDirectory) {
+      gistURI = await Service.exportToGist(fileOrDirectory, this.state.fiddle);
+    } else {
+      const projectModel = this.state.project.getModel();
```

Comment:
can you replace the projectModel var with `const target: File = fileOrDirectory || this.state.project.getModel();`?

this means you don't need the if and gistURL change.